### PR TITLE
Added Day, and WorkUnit classes for Calendar

### PR DESCRIPTION
**Pull request by BaqWin**
#201 #202 
Added basic implementation of Day and WorkUnit classes.
Deleted unnecessary text from readme.

### CODE REVIEW

File: modules/backend/calendar/WorkUnit.hpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+class WorkUnit
+{
+  public:
+    explicit WorkUnit(clinic::staff::doctor::iDoctor* doc, clinic::room::iRoom* room);
+    void addNewTimeSlot(std::unique_ptr<calendar::TimeSlot> timeSlot);
+    clinic::staff::doctor::iDoctor* getDoctor() const;
+    clinic::room::iRoom* getRoom() const;
+
+  private:
+    std::vector<std::unique_ptr<calendar::TimeSlot>> _slot_collection;
```
Comment (fbtom) [inline modules/backend/calendar/WorkUnit.hpp:22]: Follow with underscore at the end
Instead collection I'd rather see time_slots (multiple of timeslot)

File: modules/backend/calendar/WorkUnit.hpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    clinic::staff::doctor::iDoctor* doc_ptr_;
+    clinic::room::iRoom* room_ptr_;
```
Comment (fbtom) [inline modules/backend/calendar/WorkUnit.hpp:24]: not needed to mark it in name as ptr

File: modules/backend/calendar/WorkUnit.hpp [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "../clinic/room/iRoom.hpp"
+#include "../clinic/staff/doctors/iDoctor.hpp"
```
Comment (fbtom) [inline modules/backend/calendar/WorkUnit.hpp:7]: extend target_link_libraries in CmakeLists with `clinic-lib` and you should be able to use includes as follow:
`room/iRoom.hpp` `doctors/iDoctor.hpp`

File: modules/backend/calendar/WorkUnit.cpp [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace calendar
+{
+
+WorkUnit::WorkUnit(clinic::staff::doctor::iDoctor* doc, clinic::room::iRoom* room) : doc_ptr_(doc), room_ptr_(room)
+{
+}
+void WorkUnit::addNewTimeSlot(std::unique_ptr<calendar::TimeSlot> timeSlot)
+{
+    _slot_collection.push_back(std::move(timeSlot));
```
Comment (fbtom) [inline modules/backend/calendar/WorkUnit.cpp:11]: suggestion to use `emplace_back` as we might be calling createTimeSlot(...) and moving it straight to WorkUnit object

**Comment by ziobron** [comment]
Thanks @fbtom for Code Review.
🏅 1 XP granted

File: modules/backend/calendar/WorkUnit.hpp (hunk 1 above)
Comment (BaqWin) [reply modules/backend/calendar/WorkUnit.hpp:22]: done

File: modules/backend/calendar/WorkUnit.hpp (hunk 2 above)
Comment (BaqWin) [reply modules/backend/calendar/WorkUnit.hpp:24]: done

File: modules/backend/calendar/WorkUnit.hpp (hunk 3 above)
Comment (BaqWin) [reply modules/backend/calendar/WorkUnit.hpp:7]: done

File: modules/backend/calendar/WorkUnit.cpp (hunk 4 above)
Comment (BaqWin) [reply modules/backend/calendar/WorkUnit.cpp:11]: done

**Review by fbtom — APPROVED** [review]

**Comment by ziobron** [comment]
Thanks @fbtom for Code Review.
🏅 1 XP granted

**Comment by ziobron** [comment]
Your PR was merged!
🏅 2 XP granted. Thanks @BaqWin for making course materials better!

**Pull request merged**